Final tests and adjustments to allow mapping properties with hooks

beberlei opened the pull request:
Adding support for non-virtual property hooks now that native lazy objects are supported.

Property hooks are not supported when using symfony/var-exporter.

beberlei committed:
Add test for property hooks.

faizanakram99 commented:
symfony/var-exporter added support for hooked properties https://github.com/symfony/symfony/pull/59965

beberlei committed:
Merge origin/3.4.x

beberlei committed:
Tests with property hooks green when using native lazy objects.'

Also prevent using property hooks with symfony/var-exporter for now.
They did add support in a recent version, but my tests still failed with
that, so not bothering with that for now.

beberlei committed:
Add and improve property hook tests.

beberlei commented on src/Mapping/ClassMetadataFactory.php:
```diff
@@ -710,6 +701,18 @@ private function inheritIdGeneratorMapping(ClassMetadata $class, ClassMetadata $
+            return;
+        }
+
+        foreach ($class->propertyAccessors as $propertyAccessor) {
```
`ClassMetadataFactory:: validateRuntimeMetadata` is called to early, as it does not have access to any accessors yet. However this is in persistence and cannot be changed easily. 

Here its available, however at the downside of being called in every request even when loaded from cache.

beberlei committed:
Add assertion for value vs raw value representation queries against hooked properties.

beberlei committed:
Add documentation for property hooks.

greg0ire commented on docs/en/reference/basic-mapping.rst:
```diff
@@ -224,6 +226,70 @@ and a custom ``Doctrine\ORM\Mapping\TypedFieldMapper`` implementation.
+          </entity>
+        </doctrine-mapping>
+
+If you attempt to map a virtual property with `#[Column]` an exception will be thrown.
```
```suggestion
If you attempt to map a virtual property with ``#[Column]`` an exception will be thrown.
```

greg0ire committed:
Update docs/en/reference/basic-mapping.rst
